pkg/api/latest/latest.go

```diff
@@ -18,131 +18,108 @@ package latest
 
 import (
 	"fmt"
-	"sort"
-	"strings"
 
+	"k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/api/meta"
 	"k8s.io/kubernetes/pkg/api/registered"
 	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/api/v1"
 	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/sets"
 )
 
-var (
-	allGroups = GroupMetaMap{}
-	// Group is a shortcut to allGroups.Group.
-	Group = allGroups.Group
-	// RegisterGroup is a shortcut to allGroups.RegisterGroup.
-	RegisterGroup = allGroups.RegisterGroup
-	// GroupOrDie is a shortcut to allGroups.GroupOrDie.
-	GroupOrDie = allGroups.GroupOrDie
-	// AllPreferredGroupVersions returns the preferred versions of all
-	// registered groups in the form of "group1/version1,group2/version2,..."
-	AllPreferredGroupVersions = allGroups.AllPreferredGroupVersions
-	// IsRegistered is a shortcut to allGroups.IsRegistered.
-	IsRegistered = allGroups.IsRegistered
-)
+// ExternalVersions lists all allowed external versions for this group from most preferred to least preferred
+var ExternalVersions = []unversioned.GroupVersion{}
 
-// ExternalVersions is a list of all external versions for this API group in order of
-// most preferred to least preferred
-var ExternalVersions = []unversioned.GroupVersion{
-	{Group: "", Version: "v1"},
-}
+// PreferredExternalVersion is the most preferred external version (ExternalVersions[0])
+var PreferredExternalVersion = unversioned.GroupVersion{}
```

Comment:
I see, you have one of these per latest package, and you added a latest package per group. Now I understand what you were thinking.

...but we want a single latest package, not a latest package per group. Given that the latest package ought to be singular, a variable of this name doesn't make sense in it.